packages/server/utils/segmentIo.ts

```diff
@@ -9,12 +11,17 @@ const segmentIo = new SegmentIo(SEGMENT_WRITE_KEY || 'x', {
   enable: !!SEGMENT_WRITE_KEY
 }) as any
 segmentIo._track = segmentIo.track
-segmentIo.track = async (options: any) => {
+segmentIo.track = async (options: any, dataloader?: NullableDataLoader<string, IUser, string>) => {
   // used as a failsafe for PPMIs
   if (!SEGMENT_WRITE_KEY) return
   const {userId, event, properties: inProps} = options
   if (!userId) return
-  const user = await getUserById(userId)
+  /*
+   * Since segmentIo is called directly throughout server side code, we can't depend on the dataloader
+   * from analytics.ts and create a new one as needed
+   */
+  const localDataloader = dataloader ?? getDataLoader().get('users')
+  const user = await localDataloader.load(userId)
```

Comment:
This got a little messy because we call `segmentIo.track()` throughout the server side code base, so I had to create a fallback dataloader if we didn't pass one in, but I _think_ I set it up in a decent way.  I'm cleaning up all the direct segmentIo calls in a follow-on PR, so we should be able to clean up this logic at that point as well.

Reply:
+1 in almost all places where we call `segmentIo.track` we also have a dataloader available. It would be nice to pass it when calling this